tests/test_bookmarks.py

```diff
@@ -92,29 +92,38 @@ def test_run(self):
                     self.assertEqual(first_sync_bookmark_value,
                                      second_sync_bookmark_value)
 
-                    # Verify that first sync records fall betwen the start date and the final
+                    # Verify that first sync records fall between the start date and the final
                     # bookmark value
                     for message in first_sync_messages:
                         lower_bound = strptime_to_utc(self.get_properties()['start_date'])
-                        actual_value = strptime_to_utc(message.get('data').get(replication_key))
                         upper_bound = strptime_to_utc(first_sync_bookmark_value)
+                        record = message.get('data')
+                        actual_values = [strptime_to_utc(record.get(replication_key))
+                                         for replication_key in replication_keys if replication_key in record]
+
+                        self.assertNotEqual(actual_values, [], msg="No replication key found in record")
+
                         self.assertTrue(
-                            lower_bound <= actual_value <= upper_bound,
+                            any([lower_bound <= actual_value <= upper_bound for actual_value in actual_values]),
                             msg="First sync records fall outside of expected sync window"
                         )
 
                     # Verify the second sync records fall between simulated bookmark value and the
                     # final bookmark value
                     for message in second_sync_messages:
                         lower_bound = strptime_to_utc(simulated_bookmark_value)
-                        actual_value = strptime_to_utc(message.get('data',{}).get(replication_key))
                         upper_bound = strptime_to_utc(second_sync_bookmark_value)
+                        record = message.get('data')
+                        actual_values = [strptime_to_utc(record.get(replication_key))
```

Comment:
This is sort of an intereting case, since I don't think we've ever used two fields for a replication key before. This use case is a sort of `key1 OR key2` approach, rather than a `key1 AND key2` approach (like would be the case for a composite primary key).

The replication key metadata in a SaaS tap is mostly informational, so that may not be an issue, but it's an interesting piece that I hadn't considered. What folks have done in the past where the max of two fields is the replication key would be to consolidate them into a single manufactured field (like `row_updated` or something) and bookmark based on that, but this approach seems more fine grained as far as state goes, so I'm leaning towards it being good as-is. I just wanted to raise that as a consideration on my end.